File: 2021/Day_01/Python/solution.py
"""Advent of Code 2021 Day 01 Solution"""


def input_per_line(file: str):
    """This is for when each line is an input to the puzzle. The newline character is stripped."""
    with open(file, 'r') as input_file:
        return [line.rstrip() for line in input_file.readlines()]


if __name__ == "__main__":
    sonar_measurements = input_per_line("../input.txt")
    increase_count = 0
    previous_measurement = int(sonar_measurements[0])
    for index in range(1, len(sonar_measurements)):
        current_measurement = int(sonar_measurements[index])
        if current_measurement > previous_measurement:
            increase_count += 1
        previous_measurement = current_measurement
    print(f"There are {increase_count} measurements larger than the previous one.")

File: 2021/Day_01/Ruby/solution.rb
require "../../input_parsing/parse_input"

depths = input_per_line("../input.txt")
answer_one = depths.map(&:to_i).each_cons(2).map{|a, b| a < b ? 1 : 0}.reduce(:+)
puts "There are #{answer_one} measurements larger than the previous one."
answer_two = depths.map(&:to_i).each_cons(3).map{|a, b, c| a + b + c}.each_cons(2).map{|a, b| a < b ? 1 : 0}.reduce(:+)
puts "There are #{answer_two} measurements larger than the previous one when considering 3 at a time."